compiler/rustc_mir_build/src/thir/pattern/check_match.rs

```diff
@@ -960,10 +970,27 @@ fn report_unreachable_pattern<'p, 'tcx>(
 }
 
 /// Report unreachable arms, if any.
-fn report_arm_reachability<'p, 'tcx>(cx: &PatCtxt<'p, 'tcx>, report: &UsefulnessReport<'p, 'tcx>) {
+fn report_arm_reachability<'p, 'tcx>(
+    cx: &PatCtxt<'p, 'tcx>,
+    report: &UsefulnessReport<'p, 'tcx>,
+    suggest_removing_if_unreachable: bool,
```

Comment:
I wonder if we should give this a name that better explains why we'd set it to `false`... i.e. `is_match_arm` or something, since (afaict?) it's only set to false when in non-match constructs like let.